great_expectations/expectations/core/expect_query_results_to_match_source.py

```diff
@@ -296,6 +299,110 @@ def _validate(
                 },
             }
 
+    @override
+    @classmethod
+    @renderer(renderer_type=AtomicDiagnosticRendererType.OBSERVED_VALUE)
+    def _atomic_diagnostic_observed_value(
+        cls,
+        configuration: Optional[ExpectationConfiguration] = None,
+        result: Optional[ExpectationValidationResult] = None,
+        runtime_configuration: Optional[dict] = None,
+    ) -> list[RenderedAtomicContent]:
+        details = cls._get_details_from_results(result)
+
+        missing_rows: list[dict[str, Any]] = details["missing_rows"]
+        unexpected_rows: list[dict[str, Any]] = details["unexpected_rows"]
+        missing_rows_table = cls._create_observed_values_table(missing_rows)
+        unexpected_rows_table = cls._create_observed_values_table(unexpected_rows)
+
+        return [
+            cls._create_table_rendered_atomic_content(
+                unexpected_rows_table,
+                label="Unexpected records",
+            ),
+            cls._create_table_rendered_atomic_content(
+                missing_rows_table,
+                label="Missing records",
+            ),
+        ]
```

Comment:
Looks like there is going to need to be one more block above these that renders the `description` if it was set. I would add a helper that instantiates `RendererConfiguration` and get `template_str` from that for `RenderedAtomicContent`. Since we only render the `description`, you don't actually need any of the other `RendererConfiguration` logic (`include_column_name`, `row_condition` `meta_notes`, and/or evaluation parameters).

Reply:
Talked async. Updated schema types, but not doing this as the prescriptive renderer should handle this.